kv/kvclient/kvstreamer: TestStreamerVaryingResponseSizes failed

kv/kvclient/kvstreamer.TestStreamerVaryingResponseSizes [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14971871?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14971871?buildTab=artifacts#/) on release-23.2 @ [f85987da7b5b9db3015900cf6dfa37c2ccd91c99](https://github.com/cockroachdb/cockroach/commits/f85987da7b5b9db3015900cf6dfa37c2ccd91c99):

Fatal error:

```
panic: test timed out after 4m55s
running tests:
	TestStreamerVaryingResponseSizes (4m5s)

```
Stack: 

```
goroutine 78943 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x320
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStreamerVaryingResponseSizes
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/cfc927495bcc48713af623bdbb6e7a90/logTestStreamerVaryingResponseSizes65494685
    test_log_scope.go:81: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamerVaryingResponseSizes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-38163

--- a/pkg/kv/kvclient/kvstreamer/BUILD.bazel
+++ b/pkg/kv/kvclient/kvstreamer/BUILD.bazel
@@ -70,6 +70,7 @@ go_test(
         "//pkg/settings/cluster",
         "//pkg/sql",
         "//pkg/sql/rowcontainer",
+        "//pkg/sql/sem/eval",
         "//pkg/storage",
         "//pkg/testutils/serverutils",
         "//pkg/testutils/skip",

--- a/pkg/kv/kvclient/kvstreamer/streamer_test.go
+++ b/pkg/kv/kvclient/kvstreamer/streamer_test.go
@@ -30,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/concurrency/lock"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
@@ -554,7 +555,15 @@ func TestStreamerVaryingResponseSizes(t *testing.T) {
 
 	skip.UnderDuress(t)
 
-	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			SQLEvalContext: &eval.TestingKnobs{
+				// We disable the randomization of some batch sizes because with
+				// some low values the test takes much longer.
+				ForceProductionValues: true,
+			},
+		},
+	})
 	defer s.Stopper().Stop(context.Background())
 
 	runner := sqlutils.MakeSQLRunner(db)

